lib/actions/preload.coffee

```diff
@@ -0,0 +1,208 @@
+			signature: 'commit'
+			parameter: 'hash'
+			description: 'a specific application commit to preload (ignored if no appId is given)'
+			alias: 'c'
+		}
+		{
+			signature: 'splash-image'
+			parameter: 'splashImage.png'
+			description: 'path to a png image to replace the splash screen'
+			alias: 's'
+		}
+		{
+			signature: 'dont-detect-flasher-type-images'
+			boolean: true
+			description: 'Disables the flasher type images detection: treats all images as non flasher types'
+		}
+		{
+			signature: 'docker'
+			parameter: 'docker'
+			description: 'Path to a local docker socket'
+			alias: 'P'
+		}
+		{
+			signature: 'dockerHost'
+			parameter: 'dockerHost'
+			description: 'The address of the host containing the docker daemon'
+			alias: 'h'
+		}
+		{
+			signature: 'dockerPort'
+			parameter: 'dockerPort'
+			description: 'The port on which the host docker daemon is listening'
+			alias: 'p'
+		}
+		{
+			signature: 'ca'
+			parameter: 'ca'
+			description: 'Docker host TLS certificate authority file'
+		}
+		{
+			signature: 'cert'
+			parameter: 'cert'
+			description: 'Docker host TLS certificate file'
+		}
+		{
+			signature: 'key'
+			parameter: 'key'
+			description: 'Docker host TLS key file'
+		}
+	]
+	action: (params, options, done) ->
+		streamToPromise = require('stream-to-promise')
+
+		options.image = params.image
+		options.appId = options.app
+		delete options.app
+
+		options.dontDetectFlasherTypeImages = options['dont-detect-flasher-type-images']
+		delete options['dont-detect-flasher-type-images']
+
+		dockerUtils.getDocker(options)
+		.then (docker) ->
+			buildOutputStream = preload.build(docker)
+			buildOutputStream.pipe(process.stdout)
+			streamToPromise(buildOutputStream)
+			.then(resin.settings.getAll)
+			.then (settings) ->
+				options.proxy = settings.proxy
+				options.apiHost = settings.apiUrl
+				preload.getDeviceTypeSlug(docker, options)
+			.then (deviceType) ->
+				Promise.try ->
+					if options.appId
+						return resin.models.application.get(options.appId)
+					return selectApplication(resin, form, deviceType)
+				.then (application) ->
+					options.appId = application.id
+					if deviceType != application.device_type
+						expectedError(
+							"Image device type (#{application.device_type}) and application device type (#{deviceType}) do not match"
+						)
+					return resin.models.build.getAllByApplication(application.id)
+					.then (builds) ->
+						if options.commit
+							if _.map(builds, 'commit_hash').indexOf(options.commit) == -1
+								expectedError('There is no build matching this commit')
+							return options.commit
+						return selectApplicationCommit(resin, form, application.id)
+					.then (commit) ->
+						if commit != LATEST
+							options.commit = commit
+						return disableAutomaticUpdates(form, resin, application, commit)
+			.then(resin.auth.getToken)
+			.then (token) ->
+				options.apiToken = token
+				preload.run(docker, options)
+		.then (streams) ->
```

Comment:
I didn't use because I use `streams.map` below, that would require to recreate an array like `[ stdout, stderr ]`